src/protocol/attribution/aggregate_credit.rs

```diff
@@ -5,38 +5,46 @@ use super::{
     },
     InteractionPatternStep,
 };
-use crate::bits::BitArray;
 use crate::error::Error;
-use crate::ff::Field;
-use crate::protocol::basics::SecureMul;
 use crate::protocol::context::{Context, SemiHonestContext};
 use crate::protocol::modulus_conversion::split_into_multi_bit_slices;
 use crate::protocol::sort::apply_sort::apply_sort_permutation;
 use crate::protocol::sort::generate_permutation::generate_permutation_and_reveal_shuffled;
 use crate::protocol::{RecordId, Substep};
 use crate::secret_sharing::replicated::semi_honest::AdditiveShare as Replicated;
+use crate::{bits::BitArray, secret_sharing::Arithmetic};
+use crate::{ff::Field, protocol::basics::SecureMul};
 use futures::future::{try_join, try_join_all};
-use std::iter::repeat;
+use std::{iter::repeat, marker::PhantomData};
 
 /// Aggregation step for Oblivious Attribution protocol.
 /// # Panics
 /// It probably won't
 ///
 /// # Errors
 /// propagates errors from multiplications
-pub async fn aggregate_credit<F: Field, BK: BitArray>(
+pub async fn aggregate_credit<F, BK>(
     ctx: SemiHonestContext<'_, F>,
-    capped_credits: &[MCAggregateCreditInputRow<F>],
+    capped_credits: &[MCAggregateCreditInputRow<F, Replicated<F>>],
     max_breakdown_key: u128,
     num_multi_bits: u32,
-) -> Result<Vec<MCAggregateCreditOutputRow<F>>, Error> {
+    _is_malicious: bool,
```

Comment:
If this is unused - why is it here? Can we remove this?

Reply:
Yes, it is removed in next PR. Let me remove from here itself

Reply:
It's fine - I'll assume it's used in the next PR. No need to remove.